EasyPost.Tests/BetaFeaturesTests/ParametersTests/ParametersTest.cs

```diff
@@ -102,34 +102,43 @@ public void TestParametersToDictionaryWithSubDictionary()
             shipmentData = dictionary["shipment"] as Dictionary<string, object>;
             addressData = shipmentData["to_address"] as Dictionary<string, object>;
 
-            // The value of "street1" should be the value of "streetB" via the Address.Create parameter object
+            // The value of "street1" should be the value of "streetB" via the Addresses.Create parameter object
             Assert.Equal(streetB, addressData["street1"]);
         }
 
+        /// <summary>
+        /// This test proves that we can reuse the Addresses.Create parameter object,
+        /// with its serialization logic adapting to whether it is a top-level parameter object
+        /// or a nested parameter object within another parameter object.
+        ///
+        /// Notice how the paths to "street1" are different depending on whether the Addresses.Create parameter object.
+        ///
+        /// The schema for an address creation API call contains all address data wrapped in an "address" key (excluding irrelevant "verify" and "verify_strict" keys).
+        ///
+        /// The schema for a shipment creation API call does not contain this "address" key, with all address data instead wrapped inside a "to_address" key.
+        ///
+        /// Behind the scenes, the Addresses.Create parameter object is the same, but the serialization path for each property (parameter) adapts to the context in which it is used
+        /// </summary>
```

Comment:
Looove this. A single place where I can get all the details I need to know about this test.